source/utils.js

```diff
@@ -199,6 +199,21 @@ var utils = {
 
         object[last] = value;
     },
+
+    /**
+     * Returns href from <link rel='canonical'>
+     * if presents in the document
+     *
+     * @returns {String?}
```

Comment:
There's a [nullable type](https://developers.google.com/closure/compiler/docs/js-for-compiler?hl=de#types), but I misused it — the question mark should be before the type, not after.